lib/main.js

```diff
@@ -25,383 +18,349 @@ let Emitter, CompositeDisposable, Minimap, MinimapElement, MinimapPluginGenerato
    *
    * @param {*} model the model for which returning a view
    * @return {MinimapElement}
    */
-  minimapViewProvider (model) {
-    if (!Minimap) { Minimap = require('./minimap') }
-
-    if (model instanceof Minimap) {
-      if (!MinimapElement) { MinimapElement = require('./minimap-element') }
-
-      const element = new MinimapElement()
-      element.setModel(model)
-      return element
-    }
+export function minimapViewProvider (model) {
+  if (model instanceof Minimap) {
+    const element = new MinimapElement()
+    element.setModel(model)
+    return element
   }
+}
 
-  /**
+/**
    * Deactivates the minimap package.
    */
-  deactivate () {
-    if (!this.active) { return }
+export function deactivate () {
+  if (!active) { return }
 
-    this.deactivateAllPlugins()
+  deactivateAllPlugins()
 
-    if (this.editorsMinimaps) {
-      this.editorsMinimaps.forEach((value, key) => {
-        value.destroy()
-        this.editorsMinimaps.delete(key)
-      })
-    }
-
-    this.subscriptions.dispose()
-    this.subscriptions = null
-    this.subscriptionsOfCommands.dispose()
-    this.subscriptionsOfCommands = null
-    this.editorsMinimaps = undefined
-    this.toggled = false
-    this.active = false
+  if (editorsMinimaps) {
+    editorsMinimaps.forEach((value, key) => {
+      value.destroy()
+      editorsMinimaps.delete(key)
+    })
   }
 
-  getConfigSchema () {
-    return this.config
-      ? this.config
-      : atom.packages.getLoadedPackage('minimap').metadata.configSchema
-  }
+  subscriptions.dispose()
+  subscriptions = null
+  subscriptionsOfCommands.dispose()
+  subscriptionsOfCommands = null
+  editorsMinimaps = undefined
+  toggled = false
+  active = false
+}
+
+export function getConfigSchema () {
+  return config || atom.packages.getLoadedPackage('minimap').metadata.configSchema
+}
 
-  /**
+/**
    * Toggles the minimap display.
    */
-  toggle () {
-    if (!this.active) { return }
-
-    if (this.toggled) {
-      this.toggled = false
-
-      if (this.editorsMinimaps) {
-        this.editorsMinimaps.forEach((value, key) => {
-          value.destroy()
-          this.editorsMinimaps.delete(key)
-        })
-      }
-      this.subscriptions.dispose()
-    } else {
-      this.toggled = true
-      this.initSubscriptions()
+export function toggle () {
+  if (!active) { return }
+
+  if (toggled) {
+    toggled = false
+
+    if (editorsMinimaps) {
+      editorsMinimaps.forEach((value, key) => {
+        value.destroy()
+        editorsMinimaps.delete(key)
+      })
     }
+    subscriptions.dispose()
+  } else {
+    toggled = true
+    initSubscriptions()
   }
+}
 
-  /**
+/**
    * Opens the plugin generation view.
    *
    * @param  {string} template the name of the template to use
    */
-  generatePlugin (template) {
-    if (!MinimapPluginGeneratorElement) {
-      MinimapPluginGeneratorElement = require('./minimap-plugin-generator-element')
```

Comment:
It is almost certainly faster to lazy load `'./minimap-plugin-generator-element'` since most users will never use it.

Reply:
Yes, thanks. I wanted to go through the code and find these parts that are not used in the default workflow.

The previous code was making literally everything lazy even it was not! This had resulted in a lot of race conditions, and the need for applying hacks for modifying the require cache.

Another module that I found that is not needed is the LegacyAdapter

Reply:
Fixed in https://github.com/atom-minimap/minimap/pull/736/commits/c7d058dcc60d33f25138ba92ecd90777fd5e99dc